packages/story-editor/src/components/panels/design/link/link.js

```diff
@@ -201,6 +204,8 @@ function LinkPanel({ selectedElements, pushUpdateForObject }) {
     link.url,
   ]);
 
+  const linkIcon = getProxiedUrl({ needsProxy: true }, encodeURI(link.icon));
```

Comment:
It may be overkill, but we went the same route with hotlinking as well. So let's do the same thing here for consistency and to avoid sending _everything_ through the proxy.

We can move the `checkResourceAccess` function from `useInsert.js` to `useCORSProxy.js` and then use that here.